jetbrains-core/src/software/aws/toolkits/jetbrains/services/ecs/exec/EcsExecUtils.kt

```diff
@@ -87,4 +97,53 @@ object EcsExecUtils : CoroutineScope by ApplicationThreadPoolScope("EcsExec") {
             }
         )
     }
+
+    fun checkRequiredPermissions(project: Project, clusterArn: String, taskArn: String) {
+        val ecsClient = project.awsClient<EcsClient>()
+        val iamClient = project.awsClient<IamClient>()
+        val task = ecsClient.describeTasks(DescribeTasksRequest.builder().tasks(taskArn).cluster(clusterArn).build())
+        val taskRoleArn = if (task.tasks()[0].overrides().taskRoleArn() != null) {
+            task.tasks()[0].overrides().taskRoleArn()
```

Comment:
Why do we always take the first task?

Reply:
We are just checking the permissions on one task selected, so we need the details/TaskRoleArn on the first task of the DescribeTasks call

Reply:
Can each task have a different role? If so do we need to search for the task that matches the requested one by the user?

Reply:
Task Roles are attached to task definitions. Searching through numerous tasks and checking if all/any of them have required permissions might be more expensive (includes list-task, describe-task calls, followed by this function for each task) than checking for just the task selected by the user. One alternative that we can use is if the task doesn't have required permissions, we display the warning and redirect them to the same dialog to select another task.